src/main/frontend/JS/Admin.js

```diff
@@ -0,0 +1,405 @@
+                console.log(data.message);
+                fetchData(pagenumber, pagesize, sortOrder)
+                // Optionally update your UI based on the response
+            })
+            .catch(error => {
+                console.error('Error updating data:', error.message);
+            });
+
+        // Remove the modal overlay after submission
+        document.body.removeChild(modalOverlay);
+    });
+}
+
+
+function editData(button) {
+    let row = button.parentNode.parentNode;
+
+    // Create a form element
+    let form = document.createElement("form");
+    form.className = "editform"
+    let fields
+    let exitButton = document.createElement("button");
+    exitButton.textContent = "X";
+    exitButton.className = "exitbutton"
+
+    exitButton.onclick=function() {
+        document.body.removeChild(modalOverlay);
+    };
+
+    // Define the fields you want to edit
+    if (row.cells[6].innerHTML === "EHAM")
+        fields = ["Callsign", "Operator", "ICAOType", "ADEP", "DEST", "FLIGHT_RULES", "TAS", "RFL", "WeightClass", "RunWay", "GATE", "STACK"];
+    else
+        fields = ["Callsign", "Operator", "ICAOType", "ADEP", "DEST", "FLIGHT_RULES", "TAS", "RFL", "WeightClass"];
+
+    // Iterate over the fields and create textboxes for each
+    fields.forEach(field => {
+        let label = document.createElement("label");
+        label.textContent = `Enter the new ${field}: `;
+        let input = document.createElement("input");
+
+        // Adjust the cell index for RunWay and GATE
+        let cellIndex;
+        if (field === "RunWay") {
+            cellIndex = 13;
+        } else if (field === "GATE") {
+            cellIndex = 14;
+        } else if (field === "STACK") {
+            cellIndex = 15;
+        } else {
+            cellIndex = fields.indexOf(field) + 2;
+        }
+
+        input.value = row.cells[cellIndex].innerHTML;
+        input.type = "text";
+        input.name = field;
+        form.appendChild(label);
+        form.appendChild(input);
+        form.appendChild(document.createElement("br"));
+    });
+
+    // Add a submit button to the form
+    let submitButton = document.createElement("button");
+    submitButton.textContent = "Submit";
+    form.appendChild(submitButton);
+
+    // Create a modal overlay
+    let modalOverlay = document.createElement("div");
+    modalOverlay.classList.add("modal-overlay");
+
+    // Display the form in the modal overlay
+    modalOverlay.appendChild(exitButton);
+    modalOverlay.appendChild(form);
+    document.body.appendChild(modalOverlay);
+
+
+    // Handle the form submission
+    form.addEventListener("submit", function (event) {
+        event.preventDefault();
+
+        // Extract values from the form
+        let updatedData = {FLIGHT_ID: row.cells[1].innerHTML};
+        fields.forEach(field => {
+            updatedData[field] = form.elements[field].value;
+        });
+        console.log(JSON.stringify(updatedData))
```

Comment:
try to remove all `console.log()` before mergin into main. This will give a pretty decent performance increase ( depending on how mutch and what you console.log ( you should keep the console.errors and console.warn tho